src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -87,12 +87,23 @@ private static SocketAsyncEngine[] CreateEngines()
         //
         private readonly ConcurrentQueue<SocketIOEvent> _eventQueue = new ConcurrentQueue<SocketIOEvent>();
 
+        // The scheme works as following:
+        // From NotScheduled, the only transition is to Scheduled when new events are enqueued and a work item is enqueued to process them.
+        // From Scheduled, the only transition is to Determining right before trying to dequeue an event.
+        // From Determining, it can go to either NotScheduled when no events are present in the queue (the previous work item processed all of them)
+        // or Scheduled if the queue is still not empty (let the current work item handle parallelization as convinient).
```

Comment:
```suggestion
        // The scheme works as follows:
        // - From NotScheduled, the only transition is to Scheduled when new events are enqueued and a work item is enqueued to process them.
        // - From Scheduled, the only transition is to Determining right before trying to dequeue an event.
        // - From Determining, it can go to either NotScheduled when no events are present in the queue (the previous work item processed all of them)
        //   or Scheduled if the queue is still not empty (let the current work item handle parallelization as convinient).
```